nixos/tests/containers-network-namespace.nix

```diff
@@ -0,0 +1,38 @@
+import ./make-test-python.nix {
+
+  machine =
+    { config, pkgs, ... }:
+    {
+      networking.namespaces = [ "alice" "bob" ];
+
+      containers.alice = {
+        autoStart = true;
+        networkNamespace = "alice";
+        config = { ... }: {};
+      };
+
+      containers.bob = {
+        autoStart = true;
+        networkNamespace = "bob";
+        config = { ... }: {};
+      };
+    };
+
+  testScript =
+    ''
+      machine.start()
+      machine.wait_for_unit("default.target")
+
+      assert "alice" in machine.succeed("ip netns"), "alice: missing network namespace"
+      machine.succeed("ip -n alice link add alice_nic type dummy")
+      assert "alice_nic: <BROADCAST,NOARP> mtu" in machine.succeed(
+          "nixos-container run alice /run/current-system/sw/bin/ip link show alice_nic"
```

Comment:
That output might not be stable - If you want to extract things from `ip l`, you should use the `-j` option.

However, in that case we only want to check if the interface exists - checking in `/sys/class/net/$ifname` should be sufficient.

Reply:
checking in `/sys` was a little problematic from another network namespace (see: 68708349f111da4c2c9ec3b4ac6e5105796ee9c4), so I rewrote using `ip -j l`.